docs/application/dotnet/guides/applications/nui-widget-app.md

```diff
@@ -0,0 +1,244 @@
+
+## Prerequisites
+To enable your application to use the NUI widget functionality:
+
+To use the methods and properties of [Tizen.NUI](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.html) namespace, include it in your application:
+
+   ```csharp
+   using Tizen.NUI;
+   ```
+
+## Create a NUI Widget Application
+
+If you want to use NUI for your widget application, you must create the application with [NUI Widget Application](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.NUIWidgetApplication.html) class. 
+
+1. Define your widget application class, which is inherited from [NUIWidgetApplication](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.NUIWidgetApplication.html) class:
+
+   ```csharp
+   class Program : NUIWidgetApplication
+   ```
+3. The widget application starts with `Main()`, which creates and initializes the application. `Run()` method of [NUIWidgetApplication](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.NUIWidgetApplication.html) class is used to start the application event-loop.
+
+   `NUIWidgetApplication` class provides two kinds of constructors:
+   - For using the `NUIWidgetApplication(Type type)` constructor, in case the widget application's ID is the same as the application ID.
+      ```csharp
+        static void Main(string[] args)
+        {
+            var app = new Program(typeof(MyWidget));
+            app.Run(args);
+        }    
+      ```
+
+   - For using the `NUIWidgetApplication(Dictionary< Type, string > typeInfo)` constructor, in case your widget applications have multiple widget classes.
+
+      For multiple instances, add `<widget-class>` in XML as follow:
+        ```XML
+          <widget-class classid="second" update-period="0">
+            <support-size preview="Widget.png">2x2</support-size>
+          </widget-class>
+        ```
+      You can then modify the code as follows:
+        ```csharp
+        static void Main(string[] args)
+        {
+           Dictionary<System.Type, string> widgetSet = new Dictionary<Type, string>();
+            widgetSet.Add(typeof(MyWidget), "second@org.tizen.example.WidgetTemplate");
+            var app = new Program(widgetSet);
+            app.Run(args);
+        }    
+        ```
+
+4. Define your widget class, which is inherited from [Widget](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.Widget.html):
+
+   ```csharp
+   class MyWidget : Widget
+   ```
+
+   
+
+5. Override event callback methods of your new class:
+    ```csharp
+    class MyWidget : Tizen.NUI.Widget
+    {
+      protected override void OnCreate(string contentInfo, Window window)
+      {
+        /// Create the UI
+        /// ....
+        base.OnCreate(contentInfo, window);
+      }
+    }
+    ```
+
+   `Widget` class provides the following callback:
+
+    - `OnCreate()` callback is triggered when the widget instance is created.
+
+    - `OnTerminate()` callback is triggered when the widget instance is terminated.
```

Comment:
Please add colon instead of period as a code follows:

```suggestion
- `OnTerminate()` callback is triggered when the widget instance is terminated:
```